jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -359,6 +359,20 @@ static void addCompletionListener(Request request, Consumer<Throwable> listener)
      */
     Session getSession(boolean create);
 
+    /** If sessions are supported by a given configuration, return true
+     * if the request URL contains a valid session id, false otherwise.
+     * @return true if the request URL contained a valid session id, false
+     * otherwise.
+     */
+    boolean isRequestedSessionIdFromURL();
+
+    /** If sessions are supported by a given configuration, return true if
+     * the request contained a cookie with a valid session id, false otherwise.
+     * @return true if the request contains a valid session id cookie, false
+     * otherwise.
+     */
+    boolean isRequestedSessionIdFromCookie();
+
```

Comment:
We really REALLY do not want extra methods on the Request interface.     
Let me ponder a better solution...

Reply:
@gregw  I understand, and I thought this would be the case. However, we already expose `getSession(boolean)` on the core `Request` interface, so it seems reasonable to also expose these. Or, we remove `getSession(boolean)` from core `Request` and find another solution for all 3 methods, but either way, I think we have to be consistent.